packages/x-charts/src/internals/Transition.tsx

```diff
@@ -0,0 +1,97 @@
+import { timer, Timer, now, timeout } from '@mui/x-charts-vendor/d3-timer';
+
+/**
+ * A resumable transition class inspired by d3-transition.
+ * Allows for starting, and stopping and resuming transitions.
+ *
+ * The transition is started automatically.
+ * A transition cannot be restarted after it has finished.
+ * Resuming a transition will continue from the point it was stopped, i.e., easing will continue from the point it was
+ * stopped.
+ */
+export class Transition {
```

Comment:
What is the difference/advantage of this over react-transition-group? 🤔

Reply:
I wasn't aware of `react-transition-group`, but from a quick glance it looks like `react-transition-group` exposes components. This `Transition` class isn't tied to React. The `Transition` component from `react-transition-group` could potentially be an alternative implementation to our `useAnimate` hook, but it seems that the `Transition` component will re-render instead of applying changes directly through a ref, which isn't ideal. 

Do you have anything particular from `react-transition-group` that you think is useful for this use case?

Reply:
Nope, I mistook what this class does 😓